mirrord-layer/src/connection.rs

```diff
@@ -106,10 +110,13 @@ pub(crate) async fn connect(config: &LayerConfig) -> impl AsyncWrite + AsyncRead
                 info!("No existing agent, spawning new one.");
                 let agent_port: u16 = rand::thread_rng().gen_range(30000..=65535);
                 info!("Using port `{agent_port:?}` for communication");
-                let pod_agent_name = match k8s_api.create_agent(agent_port).await {
-                    Ok(pod_name) => pod_name,
-                    Err(err) => handle_error(err),
-                };
+                let pod_agent_name = tokio::time::timeout(
+                    Duration::from_secs(config.agent.communication_timeout.unwrap_or(30) as u64),
```

Comment:
A bit reluctant to use the communication timeout as it's explicitly marked as agent communication timeout, and if someone has a cluster that takes more time to startup our agent they would also affect the time it takes for the agent to close after no more connections exist.
I'd add a new configuration or use a const for now (also I'd default it at around 60sec)